src/server/scripts/Northrend/Ulduar/Ulduar/ulduar.cpp

```diff
@@ -502,11 +410,98 @@ class go_call_tram : public GameObjectScript
     }
 };
 
+class spell_ulduar_teleporter : public SpellScriptLoader
+{
+public:
+    spell_ulduar_teleporter() : SpellScriptLoader("spell_ulduar_teleporter") { }
+
+    enum Teleport_Spells
+    {
+        BASE_CAMP = 64014,
+        GROUNDS = 64032,
+        FORGE = 64028,
+        SCRAPYARD = 64031,
+        ANTECHAMBER = 64030,
+        WALKWAY = 64029,
+        CONSERVATORY = 64024,
+        MADNESS = 64025,
+        SPARK = 65042
+    };
+
+    class spell_ulduar_teleporter_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_ulduar_teleporter_SpellScript);
+
+        SpellCastResult CheckRequirement()
+        {
+            if (GetExplTargetUnit()->GetTypeId() != TYPEID_PLAYER)
+                return SPELL_FAILED_DONT_REPORT;
+
+            if (GetExplTargetUnit()->IsInCombat())
+            {
+                Spell::SendCastResult(GetExplTargetUnit()->ToPlayer(), GetSpellInfo(), 0, SPELL_FAILED_AFFECTING_COMBAT);
+                return SPELL_FAILED_AFFECTING_COMBAT;
+            }
+
+            return SPELL_CAST_OK;
+        }
+
+        //doing real teleporting because all spells are just broken
+        void TeleportPlayer()
+        {
+            auto SpellID = GetSpellInfo()->Id;
+            auto player = (Player*)GetExplTargetUnit();
```

Comment:
Don't use a C style cast please. static cast or reinterpret cast, based on whatever your plan is with it. The auto keyword seems weird here, the functions that get the type aren't explicit in what they're returning in terms of type.

See here: http://www.acodersjourney.com/2016/02/c-11-auto/